components/admin/reports/GenerateReportModal.tsx

```diff
@@ -39,6 +47,99 @@ export default function GenerateReportModal<T>(props: GenerateReportProps<T>) {
   const [endDate, setEndDate] = useState('');
   const [step, setStep] = useState(GenerateReportStep.SelectColumns);
 
+  const [selectedColumns, setSelectedColumns] = useState<Set<ApplicationsReportColumn>>(new Set());
+
+  // A function that returns an event-handler function that adds/removes the column from selectedColumns
+  const handleSelectColumn =
+    (column: ApplicationsReportColumn) => (event: ChangeEvent<HTMLInputElement>) => {
+      const updatedSelectedColumns = new Set([...selectedColumns]);
+      event.target.checked
+        ? updatedSelectedColumns.add(column)
+        : updatedSelectedColumns.delete(column);
+      setSelectedColumns(updatedSelectedColumns);
+    };
+
+  const handleSelectAllColumns = (event: ChangeEvent<HTMLInputElement>) => {
+    const updatedSelectedColumns = event.target.checked
+      ? new Set([
+          ApplicationsReportColumn.UserId,
+          ApplicationsReportColumn.ApplicantName,
+          ApplicationsReportColumn.ApplicantDateOfBirth,
+          ApplicationsReportColumn.AppNumber,
+          ApplicationsReportColumn.ApplicationDate,
+          ApplicationsReportColumn.PaymentMethod,
+          ApplicationsReportColumn.FeeAmount,
+          ApplicationsReportColumn.DonationAmount,
+          ApplicationsReportColumn.TotalAmount,
+        ])
+      : new Set([]);
+    setSelectedColumns(updatedSelectedColumns);
+  };
+
+  // Toast message
+  const toast = useToast();
+
+  // Export csv query
+  const [exportCSV, { loading }] = useLazyQuery<
+    GenerateApplicationsReportResponse,
+    GenerateApplicationsReportRequest
+  >(GENERATE_APPLICATIONS_REPORT_QUERY, {
+    onCompleted: data => {
+      if (data?.generateApplicationsReport.ok) {
+        toast({
+          status: 'success',
+          description: `A CSV ${
+            page === 'permitHolders' ? 'permit holders' : 'requests'
+          } report has been successfully generated.`,
```

Comment:
Updated toast description to include report type based on design feedback https://uwblueprint.slack.com/archives/C021S4WGHBR/p1639021500001300?thread_ts=1638677484.001200&cid=C021S4WGHBR

Reply:
Great! Thanks!